### ISSUE TITLE
[BUG] Search for fields across associations

### ISSUE THREAD

**Issue by anhnk** [aesmail/kaffy#324]
**Versions Used**
Kaffy: 0.10.3
Phoenix: 1.7.18
Elixir: 1.14.1

**What's actually happening?**

An error is raised when association is used in search fields. It looks like the query builder didn't use the association at all.

I've narrowed down to the [code here](https://github.com/aesmail/kaffy/blob/master/lib/kaffy/resource_query.ex#L162-L169) in `resource_query` but couldn't make any further progress.

This is the error:
```
Postgrex.Error at GET /admin/user/post
ERROR 42703 (undefined_column) column p0.username does not exist

    query: SELECT p0."id", p0."user_id", p0."inserted_at", p0."updated_at" FROM "posts" AS p0 LEFT OUTER JOIN "users" AS u1 ON u1."id" = p0."user_id" WHERE (p0."username"::varchar ILIKE $1) ORDER BY p0."id" DESC LIMIT $2 OFFSET $3

    hint: Perhaps you meant to reference the column "u1.username".
```

This is my code:
```elixir
defmodule Mode.Kaffy.PostAdmin do
   def search_fields(_schema) do
    [
      :id,
      user: [:username]
    ]
  end
```

**What should happen instead?**

The Query builder should use association for search field with association

**Screenshots**

**Comment by anhnk** [comment]
I downgraded to v0.10.2 and it worked. It's possible that the bug was introduced in [this commit](https://github.com/aesmail/kaffy/commit/f1dea1dc73bfb5ee8f43187848b6421e6b2c81df).

**Comment by ghenry** [comment]
Apologies for this. Will have a look. If you a PR, that would be much
appreciated. Then we can do a new release.

**Comment by anhnk** [comment]
Thanks for the prompt response! I've created [a PR](https://github.com/aesmail/kaffy/pull/325). 

Looking forward to pulling the latest version. Kaffy has been saving me lots of time. Huge kudos to the team!

**Comment by anhnk** [comment]
Thanks for merging the PR! Please let me know when you will be able to release a new version for the hex package.

**Comment by anhnk** [comment]
@ghenry May I ask how often you guys release a new version?

**Comment by ghenry** [comment]
Just whenever. Need one? We're kind of in just keep things ticking over mode.

**Comment by anhnk** [comment]
Yes, please. I'd love to get this bug fixed. You already merged the PR.
